### PR TITLE
Added sample apps for creating a home/shell app for IoT core

### PR DESCRIPTION
**Pull request by javiddhankwala**
Added two apps that uses two different approaches - via protocol handler and via restricted API.

### CODE REVIEW

File: IoTHomeAppSample/IoTHomeApp/MainPage.xaml.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+void MainPage::OemApp1_Click(Platform::Object^ sender, Windows::UI::Xaml::RoutedEventArgs^ e)
+{
+    //Launches the app that has registered as a protocol handler for 'oemapp1' protocol
+    //Use Launcher option to specify the Target Application Package Family Name to resolve
+    //the ambiguity, in case, more than one app has registered for the same protocol
+    //More information about the api can be found at:
+    //https://msdn.microsoft.com/en-us/library/windows/apps/hh701484.aspx
+
+    auto options = ref new LauncherOptions();
```
Comment (bfjelds) [inline IoTHomeAppSample/IoTHomeApp/MainPage.xaml.cpp:38]: why is the package family name provided here, but not for OemApp2_Click?

File: IoTHomeAppSample/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+Demonstrates how one can use LaunchUriAsync to launch other apps registered for a 'protocol'.
+The solution folder contains OemApp1 and OemApp2 projects that supports protocol 'oemapp1' and 'oemapp2'.
+The app also shows how to resolve ambiguity, in case of multiple apps registered for a protocol, by providing the App Id of the target application in the Launcher options.
+
+OemApp1 amd OemApp2 must be built and deployed before one can use IotHomeApp to launch them.
+
+## IoTStartApp
+
+IotStartApp illustrates how one can use PackageManager APIs to enumerate and launch apps installed on the system, just like a Start screen.
+In order to use the PackageManager APIs, it uses a restricted capability. This may make it difficult to pass the Store certification process.
```
Comment (bfjelds) [inline IoTHomeAppSample/README.md:17]: Is this true? "may make it difficult to pass" ... or does it make it impossible?

File: IoTHomeAppSample/IoTHomeApp/Package.appxmanifest [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Package xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10" xmlns:mp="http://schemas.microsoft.com/appx/2014/phone/manifest" xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10" IgnorableNamespaces="uap mp">
+  <Identity Name="ShellHomeApp" Publisher="CN=MSFT" Version="1.0.0.0" />
+  <mp:PhoneIdentity PhoneProductId="6db05a29-0188-44f7-bac0-cba49784c9d3" PhonePublisherId="00000000-0000-0000-0000-000000000000" />
+  <Properties>
```
Comment (bfjelds) [inline IoTHomeAppSample/IoTHomeApp/Package.appxmanifest:5]: you might want to demonstrate how to get back to the IoTHomeApp from App1 and App2 ... could be accomplished by adding a protocol hander here that the other apps call.

File: IoTHomeAppSample/IoTHomeApp/MainPage.xaml.cpp (hunk 1 above)
Comment (javiddhankwala) [inline IoTHomeAppSample/IoTHomeApp/MainPage.xaml.cpp:38]: This is to demonstrate the use of optional LauncherOptions, to resolve ambiguity, if multiple apps support the same protocol

File: IoTHomeAppSample/README.md (hunk 2 above)
Comment (javiddhankwala) [inline IoTHomeAppSample/README.md:17]: I thought there is some kind of white list with the store for the oems that may allow restricted capability

File: IoTHomeAppSample/IoTHomeApp/Package.appxmanifest (hunk 3 above)
Comment (javiddhankwala) [inline IoTHomeAppSample/IoTHomeApp/Package.appxmanifest:5]: Its a sample and we do demonstrate how to launch apps based on protocol.

File: IoTHomeAppSample/README.md (hunk 2 above)
Comment (bfjelds) [inline IoTHomeAppSample/README.md:17]: maybe add the oem detail

**Pull request merged**